internal/collector/processor/idbatcher/id_batcher.go

```diff
@@ -120,7 +119,7 @@ func (b *batcher) AddToCurrentBatch(id ID) {
 }
 
 func (b *batcher) CloseCurrentAndTakeFirstBatch() (Batch, bool) {
-	for readBatch := range b.batches {
+	if readBatch, ok := <-b.batches; ok {
```

Comment:
Was it intended to change the `for` by `if`?

Reply:
Yes, otherwise either test or static check would fail. The updated code is equivalent to the previous one.